.github/workflows/R-CMD-CHECK-renv.yml

```
@@ -0,0 +1,44 @@
+# This GHA checks rOpenScPCA using the repository's renv environment which records
+#  dependencies from Bioconductor 3.19, the version we expect rOpenScPCA to be most
+#  commonly used with
+
+on:
+  pull_request:
+    branches:
+      - main
+      - feature/*
+
+name: R-CMD-CHECK-renv
+
+jobs:
+  R-CMD-check-renv:
+    runs-on: ubuntu-22.04
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Set up R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: "renv"
+          use-public-rspm: true
+
+      - name: Install additional system dependencies
+        run: |
+          sudo apt-get install -y libcurl4-openssl-dev libglpk40
+
+      - name: Set up renv and install packages
+        uses: r-lib/actions/setup-renv@v2
+
+      - name: Install additional dependencies needed to check the package
+        uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rcmdcheck
+          needs: check
```

Are you sure this will not overwrite dependencies? In scpcaTools we did this more directly, as below: 
```suggestion
- name: Install additional dependencies for testing
        run: |
          install.packages("rcmdcheck")
        shell: Rscript {0}
```

---

Well it seems to have worked! I don't have any particular feelings about this step though, if you prefer using `install.packages`

---

It takes twice as long to reinstall, and isn't necessarily using the locked packages. It might by chance, and it does preserve the BioC version. Not really about a preference as much as doing what you expect.

---

> Update: it seems to be reinstalling.

Oh, I misread this as "installing", back on track!